### PR TITLE
WIP: Batch processing of individual msg types

### PR DESCRIPTION
**Pull request by Georacer**
Okay, it took the the whole evening, but I think I got something.

Comments inline, start from LogMsgGroup.

This is a work in progress, do not merge.

66.BIN finishes in 10 secs

### CODE REVIEW

**Comment by Georacer** [comment]
ReadLogLine(), findMsgStart(), readLogData() and trimTail() are no longer needed. Can we delete them?

**Comment by Georacer** [comment]
One more thing: I'm worried that typecast may mess up data conversion due to little/big-endian difference between data.

In case we see odd figures, we should keep that in mind.

**Comment by Georacer** [comment]
Just parsed the whole 160MB 51.BIN. Took 63.1 secs ^^

**Comment by Georacer** [comment]
There is definitely something wrong with encoding using typecast now. I'll investigate more.

**Comment by Georacer** [comment]
There is definitely something wrong with the exported data. They just don't make sense. I'll try to investigate more.

**Comment by Georacer** [comment]
Okay, I think the last commit fixed it.
What was happening was that I was building an array for each field
A = [msg1_byte1 msg1_byte2...; msg2_byte1 msg2_byte2...] etc
and wanted to concatenate all rows in a single row, to give it to typecast (whcih accepts vectors only)
However, I used A(:) to do that, which gives
ans = [msg1_byte1 msg2_byte1...

File: Ardupilog.m [hunk 1]
```diff
@@ -83,52 +84,136 @@ function delete(obj)
...
-                    disp('Reached end of log.')
-                    obj.numMsgs = obj.lastLineNum;
-                    break
+            % Discover the locations of all the messages
+            FMTLength = 89;
+            FMTIndices = obj.discoverMSG(128,FMTLength);
+            % Read the FMT message
+            % Generate the index mask which corresponds to the places where
+            % FMT information exists
+            indices = reshape(ones(length(FMTIndices),1)*(3:(FMTLength-1)) + (ones(length(FMTIndices),1).*FMTIndices')*ones(1,FMTLength-3),[1 length(FMTIndices)*(FMTLength-3)]);
```
Comment (hunt0r) [inline Ardupilog.m]: This line is hard for me to parse and understand. If it's okay with you, I might break it into a few lines or add more comments in a future commit.

**Comment by Georacer** [comment]
Okay, the latest commit fleshed the procedure out a bit. Thankfully there doesn't seem to be a hit in performance.

**Comment by Georacer** [comment]
Stored header indices so that we don't need to search the whole log for each message.
This led to 80% decrease in time.
Now the 51.BIN file is parsed in 15 secs!

**Comment by hunt0r** [comment]
Excellent work!

**Comment by Georacer** [comment]
Can I delete the now unused functions?

**Comment by hunt0r** [comment]
Yes. I think you can also pull out the large commented block, and any other relics from the previous approach.

Some minor ideas to consider:
1) Is it worth it to functionalize the copied portions of code for organizing and then storing the message data? If I'm not mistaken we call it in one spot for FMT messages, and identically again for the rest of the messages.
2) Do you know the reshape() function? This is more natural to me than ones()*[values] for forming a matrix of repeated values, but does the same thing. (Not worth changing if you are used to the ones()*[values] syntax, I figured it out with a pen and paper)

**Comment by Georacer** [comment]
I used reshape a few times in this PR. But I can't think of a way to use it to replace ones(r,c), when I need to create some arrays here and there.

Anyway, I'll create another PR for the cleanup tomorrow or Monday.
Then you can suggest changes if you want.

I'll also add another PR afterwards for choosing the desired message set at initializaiton.

**Comment by hunt0r** [comment]
Oh damn! Sorry, I meant "repmat()"  I also didn't mean for my multiplication signs to start/stop italics... I gotta get better at github. (I must have had reshape on my mind cause you used it)

**Pull request merged**